### ISSUE TITLE
Failed to allocate memory for custom tag binary object (-1 elements of 1 bytes each)

### ISSUE THREAD

**Issue by mokko** [python-pillow/Pillow#5314]
```python
from PIL import Image

im = Image.open('3811353.B 225 Khocho.tif')
if im.width > im.height:
    ratio = 3508 / im.width
else:
    ratio = 3508 / im.height
new_size = round(im.width * ratio), round(im.height * ratio)
if ratio < 1:  # only shrink, dont inflate
    im.thumbnail(new_size)
    im.save("new.tif")
```
Should be a simple resize operation, but I get this error:
```
new.tif: Failed to allocate memory for custom tag binary object (-1 elements of 1 bytes each).
Traceback (most recent call last):
  File "C:\Users\Win10 Pro x64\eclipse-workspace\tiffinder\resize_DinA4.pillow.py", line 20, in <module>
    im.save("new.tif")
  File "C:\Users\Win10 Pro x64\AppData\Local\Programs\Python\Python39\lib\site-packages\PIL\Image.py", line 2164, in save
    save_handler(self, fp, filename)
  File "C:\Users\Win10 Pro x64\AppData\Local\Programs\Python\Python39\lib\site-packages\PIL\TiffImagePlugin.py", line 1624, in _save
    e = Image._getencoder(im.mode, "libtiff", a, im.encoderconfig)
  File "C:\Users\Win10 Pro x64\AppData\Local\Programs\Python\Python39\lib\site-packages\PIL\Image.py", line 462, in _getencoder
    return encoder(mode, *args + extra)
RuntimeError: Error setting from dictionary
```

### What are your OS, Python and Pillow versions?

* OS:  Windows 10
* Python: 3.9.2
* Pillow: 8.1.1 and 8.1.2

Any help appreciated. Thank you!

**Comment by radarhere** [comment]
Could you provide a copy of the image?

**Comment by mokko** [comment]
https://1drv.ms/u/s!AuQGDGj3o5kXh5x1atju6i2en-2P5Q?e=Hpaauj

**Comment by radarhere** [comment]
If you're interested in an immediate fix, I find that if I replace
```python
from PIL import Image
```
with
```python
from PIL import Image, TiffTags
TiffTags.TAGS_V2[33723] = TiffTags.TagInfo("IptcNaaInfo", TiffTags.BYTE, 0)
```
then the image saves.

**Comment by mokko** [comment]
Thank you. That helps! 

I have the impression that Pillow needs to better document the metadata stuff perhaps before the code can improve. All code snippets I found on the web on exif and iptc seem to be outdated. At least that is my impression.

Let me know if I can help.

**Comment by radarhere** [comment]
I would like to add your image to the test suite, to be distributed under the Pillow license, so that we can ensure that once your error is fixed, it doesn't re-appear.

However, because your image is 40mb, I've scaled it down using macOS Preview, to a 1px image. It's attached here.
[tag_33723.tif.zip](https://github.com/python-pillow/Pillow/files/6121602/tag_33723.tif.zip)

Is it ok with you if I add that?

**Comment by mokko** [comment]
yes. Thanks for asking!

**Comment by radarhere** [comment]
On further inspection, maybe don't use the workaround I suggested above. When running tests for my fix, I found that it triggered an error under valgrind - https://github.com/radarhere/Pillow/runs/2094439293?check_suite_focus=true#step:5:2317

**Comment by wiredfool** [comment]
Great that Valgrind caught that, but ... it's supposed to only run on .c/.h changes, and you changed a .py.

**Comment by radarhere** [comment]
Testing, I find this was fixed by #5425
